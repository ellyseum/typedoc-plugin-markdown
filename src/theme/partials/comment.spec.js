describe(`comment.hbs`, () => {
  test('should compile with tags', () => {
    const context = {
      comment: {
        shortText: '',
        text: '',
        tags: [
          { tagName: 'name', paramName: '', text: 'AbstractMetadataModule' },
          {
            tagName: 'description',
            paramName: '',
            text:
              '\nProvides the module for the [[BaseClass]]\n```json\n{\n "declarations": ["AbstractMetadataComponent"],\n "exports": ["AbstractMetadataComponent"],\n "imports": [\n   "IonicModule",\n   "StackIonSelectModule"\n ],\n "providers": []\n}\n```\n',
          },
        ],
        hasVisibleComponent: () => true,
      },
    };
    const result = global.compileTemplate('comment.hbs', context);
    expect(result).toMatchSnapshot();
  });

  test('should compile with deprecation', () => {
    const context = {
      comment: {
        shortText: 'Some deprecation',
        text: '',
        tags: [
          {
            tagName: 'deprecated',
            paramName: '',
            text: '\n```typescript\nsomethingElse();\n```\n',
          },
        ],
        hasVisibleComponent: () => true,
      },
    };
    const result = global.compileTemplate('comment.hbs', context);
    expect(result).toMatchSnapshot();
  });
});
