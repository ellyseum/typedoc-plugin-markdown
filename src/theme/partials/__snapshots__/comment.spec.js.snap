// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comment.hbs should compile with deprecation 1`] = `
"Some deprecation

*__deprecated__*: 
\`\`\`typescript
somethingElse();
\`\`\`"
`;

exports[`comment.hbs should compile with tags 1`] = `
"*__name__*: AbstractMetadataModule

*__description__*: 
Provides the module for the [[BaseClass]]
\`\`\`json
{
 \\"declarations\\": [\\"AbstractMetadataComponent\\"],
 \\"exports\\": [\\"AbstractMetadataComponent\\"],
 \\"imports\\": [
   \\"IonicModule\\",
   \\"StackIonSelectModule\\"
 ],
 \\"providers\\": []
}
\`\`\`"
`;
